---
title: "Some_statistics"
author: "Jeronimo Miranda"
date: "`r Sys.Date()`"
output: github_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r source data, message=FALSE, warning=FALSE, include=FALSE}
library(ggExtra)
source("scriptlncrna.R")
source("1_someStatistics_backgroundScript.R")
```

## Transcript classification
We took the information of which mRNAs are protein coding from Trinotate, that makes a blastn and tblastn against the uniprot database to find orthologs. Protein coding are all of the mRNAs that had an ortholog in uniprot. 

```{r Coding and non coding transcripts, echo=FALSE, message=FALSE, warning=FALSE}
data_donut <- transcript_classification %>% ungroup %>% group_by(rna_type) %>%
  summarise(rna_type = unique(rna_type), number = n()) %>%
  mutate(category = case_match(rna_type, "Nonsense" ~ "Nonsense", "nonCoding" ~ "Non coding", .default = "Peptide"), label = paste0(category, "\n value: ", number), ymax = cumsum(number), ymin = lag(ymax, default = 0))

transcript_classification %>% ggplot(aes(x = 3.5, fill = rna_type)) + geom_bar() + coord_polar(theta = "y") + xlim(c(2,4)) + geom_label(data = data_donut, x =3.5, aes(y = number, label = label)) + theme_void()

```


## Length distribution of transcripts



These sequences were then aligned back to the whole transcriptome with blastn. The resulting alignments were filtered to be plus/minus alignments and to span longer than 300 bp. The Target transcripts thus found are labelled as antisense transcripts, because they match the protein coding mRNAs at least partially on their antisense strand. Out of the 39,844 unique transcripts with a swissprot annotation, we find only 9930 with at least one antisense transcript. On the other hand, the total number of unique antisense transcripts found this way is 26,953.

```{r echo=FALSE, message=FALSE, warning=FALSE}

```


The length distribution of the two differ, with the antisense transcripts being closer to the whole transcriptome numbers (i. e. skewing towards smaller). 

```{r summary, message=FALSE, warning=TRUE}
select(ampirr, total_length.qer, total_length.target) %>% transmute(Protein_coding_mRNAs = total_length.qer, Antisense_Transcripts = total_length.target) %>% summary()
```
We can see that through all quartiles, Protein coding mRNAs are longer than antisense Transcripts, with the median protein coding mRNA being 1kb while half of antisense transcripts are shorter than 413 nucleotides (this is after already filtering all non coding RNAs shorter than 200 bp). Compare to the total length distribution of the transcriptome: 75% of all transcripts have a length of less than 687 nucleotides:

```{r summary whole, echo=FALSE, message=FALSE, warning=FALSE}
summary(longitudes)
```



```{r Grafica de longitud, message=FALSE, warning=FALSE}
#pivot to longer dataframe
ampirr %>% transmute(qseqid, sseqid, Protein_coding_mRNAs = total_length.qer, Antisense_Transcripts = total_length.target) %>% 
  pivot_longer(cols = Protein_coding_mRNAs:Antisense_Transcripts, names_to = "mRNA_type", values_to = "mRNA_length") %>% 
  #This operation is to chose the transcript id from the two seqid columns
  transmute(transcript_id = if_else(mRNA_type == "Protein_coding_mRNAs", qseqid, sseqid), mRNA_type, mRNA_length) %>% unique() %>% 
  #now you can plot
  ggplot(aes(x = mRNA_length)) + geom_histogram(binwidth = 200) + scale_x_continuous(n.breaks = 10) + xlim(c(0,16500)) + theme_bw() + facet_wrap(~mRNA_type, nrow = 2, scales = "free_y") + labs(x = "mRNA length [nucleotides]", title = "Antisense transcripts are generally shorter than their targets, but more numerous", subtitle = "Histogram distribution of transcript length")

```



