---
title: "Some_statistics"
author: "Jeronimo Miranda"
date: "`r Sys.Date()`"
output: github_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r source data, message=FALSE, warning=FALSE, include=FALSE}
library(ggExtra)
library(gridExtra)
source("scriptlncrna.R")
source("1_someStatistics_backgroundScript.R")
```

## Transcript classification
We took the information of which mRNAs are protein coding from Transdecoder. More than 400,000 out of the 464,338 transcripts do not code for a detectable peptide. Afterwards, to have an idea of annotation, we use TRinotate that aligns with blastn and tblastn against the uniprot database to find putative orthologs. There is also a small but significant population of transcripts that have a significant match at the RNA level but for which transdecoder could not find a long enough ORF. We assume for now that these are either artifactual or non-functional isoforms and are marked as *"nonsense"*. Later, if a gene has at least one nonsense transcript, and no transcripts with coding potential, we will classify this gene as a pseudogene.

```{r Coding and non coding transcripts, echo=FALSE, message=FALSE, warning=FALSE}
data_donut <- transcript_classification %>% ungroup %>% group_by(rna_type) %>%
    summarise(rna_type = unique(rna_type), number = n()) %>%
    mutate(category = case_match(rna_type, "Nonsense" ~ "Nonsense", "nonCoding" ~ "Non coding", .default = "Peptide"), label = paste0(category, "\n n = ", number), ymax = cumsum(number), ymin = lag(ymax, default = 0))

transcript_classification %>% ggplot(aes(x = 3.5, fill = rna_type)) + geom_bar() + xlim(c(2,4)) + geom_label(y = c(60000, 12500, 200000), data = data_donut, x =4.1, aes(label = label)) + theme_void() + coord_polar(theta = "y") + theme(legend.position = "none") + labs(title = "The vast majority of transcripts do not code for peptides")

```

### Annotation guided more granular classification of transcripts

We have already classified non-coding transcripts using the Transdecoder annotation as support: transcripts that could be annotated just as non-coding are classified as non-sense, because they do show some similarity to uniprot transcripts, even when they do not seem to code for a peptide. The previous classification was done at the transcript level. The next classification, pertains to peptides and so, will only apply to the 60,247 transcripts for which transdecoder found at least one peptide. The relationship between transcripts and peptides is one-to-many, though, because one transcript has six possible reading frames and could even code for more than one peptide in the same reading frame. This is reflected in the 70,572 unique prot_id's from the transdecoder file.

```{r coding transcripts, echo=FALSE, message=FALSE, warning=FALSE}
data_donut <- trinotate %>% filter(!is.na(peptide_type)) %>% group_by(peptide_type) %>%
    summarise(peptide_type = unique(peptide_type), number = n()) %>%
    mutate(label = paste0(peptide_type, "\n n = ", number), ymax = cumsum(number), ymin = lag(ymax, default = 0))

trinotate %>% filter(!is.na(peptide_type)) %>% ggplot(aes(x = 3.5, fill = peptide_type)) + geom_bar(position = position_stack(reverse = TRUE)) + coord_polar(theta = "y") + xlim(c(2,4)) + theme_void() + geom_label(data=data_donut, aes(y = (ymax + ymin)/2, x = 3.5, label = label)) + theme(legend.position = "null") + labs(title = "Sizeable fractions of peptides on the wrong strand and \n peptides not present in uniprot", subtitle = "Peptide-wise classification of the ORFs predicted by transdecoder")
```

There are 28 thousand peptides with a protein match to uniprot. This is somewhat on the ballpark of the human genome? On the other hand, 31% of the peptides are predicted to be on the opposite strand (-). This is a lot, specially given our dataset strandedness. These transcripts could correspond to antisense transcripts, though. Obviously, even if transdecoder predicted ORFs on the opposite strand, if the assembly for these transcripts is correct, the peptides cannot be produced. We classify as *"Truncated"* the 3714 RNAs that code for a peptide, and show similarity at the level of the RNA but not enough to show similarity at the tblastn level. These transcripts could be added to the possible pseudogene category, although of course this is not necesarilly the case. 

Perhaps the most interesting subset are the 15737 peptides that show no similarity to any uniprot protein or mRNA. These *"Unnanotated"* dataset could correspond to novel mollusk or gasteropod genes, however, we will wait for the gene-level integration before making this call.

### Gene level integration

First, because of the one-to-many relationship between transcripts and peptides, we made a further classification of transcripts according to the peptides they produce. A transcript can code for many peptides, but we classify a transcript as *Swissprot* annotated if at least one of the peptides has a valid top ortholog from the SWissprot database. Next, a transcript is annotated as either Truncated, Unnanotated or Wrong strand if any of the peptides they produce belong to one of these categories and none of the peptides they produce belong to the others. Transcripts that produce a combination of these are not shown.

Next, we group transcripts by their Trinity assembly "gene" and classify them as:
- "Swissprot" genes if **any** of the transcripts produce a peptide with a Swissprot ortholog.
- We classified them as *Unnanotated* if **all** of the transcripts code for an unnanotated protein (this might be an underestimation).
- We classify a transcript as *Non coding* if **all** of their transcripts are non coding
- We classify a transcript as pseudogene if **all** of their transcripts belong to either non coding, truncated or 

```{r gene level, echo=FALSE, fig.height=4, fig.width=11, message=FALSE, warning=FALSE}
data_donut <- gene_classification %>% filter(gene_type != "Non coding") %>% group_by(gene_type) %>%  summarise(type = unique(gene_type), number = n()) %>% mutate(label = paste0(gene_type, "\n n = ", number), ymax = cumsum(number), ymin = lag(ymax, default = 0))

p2 <- gene_classification %>% filter(gene_type != "Non coding") %>% ggplot(aes(x=3.5, fill = gene_type)) + geom_bar(position = position_stack(reverse = T)) + coord_polar(theta = "y") + xlim(c(2,4)) + theme_void() + labs(title = "Putative pseudogenes and unnanotated novel ORFs") + geom_label(data = data_donut, x =4.1, aes(y = (ymax + ymin)/2, label = label)) + theme_void() + coord_polar(theta = "y") + theme(legend.position = "none")

p1 <- gene_classification %>% ggplot(aes(x=3.5, fill = gene_type)) + geom_bar() + coord_polar(theta = "y") + xlim(c(2,4)) + theme_void() + labs(title = "Most genes produce only Non coding transcripts", caption = "Total of 357,706 trinotate \"genes\"")

grid.arrange(p1,p2, ncol =2)
```


## Length distribution of transcripts

We will now go into length distribution of transcripts. As stated before, there are ~464,000 transcripts. By looking at a summary, we already see that the distribution is heavily right skewed. The minimum size is 169, and the maximum is 30kb (Spoiler: this is the Titin gene, the longest known protein in humans). Still, the third quartile is 687. That is, 75% of transcripts are shorter than the mean.

```{r summary of lengths, echo=TRUE, message=FALSE, warning=FALSE}
summary(longitudes$total_length)
```

To visualize this distribution, we log transform the data (exp = e)

```{r histogram whole transcriptome, echo=FALSE, message=FALSE, warning=FALSE}
longitudes %>% ggplot() + geom_histogram(aes(x = log(total_length))) + theme_bw() + labs(x = "log(transcript length) [nucleotides]", title = "The distribution of transcript length is heavily righ skewed", subtitle = "Histogram of log transformed transfript length")
```

Remember that our first classification was, transcript-wise by their coding potential, differentiating the peptide producing from the non-coding, and a third smaller group of transcripts that have a blastx hit but  do not contain an ORF long enough to match by tblastn. There is a very clear difference in the length distribution of transcripts producing a peptide in transdecoder: almost all of them are longer than the transcriptome median (log(389) = 5.95). There is also a sort of bimodal distribution in the log transformed length data for the ORF containing transcripts that could be explored later. Overall, we can conclude that a good portion of the non-coding transcriptome is shorter than 400 nucleotides.


```{r log length by coding potential, fig.height=6, fig.width=12, echo=FALSE, message=FALSE, warning=FALSE}
transcript_classification %>% ggplot() + geom_histogram(aes(x = log(total_length))) + theme_bw() + labs(x = "log(transcript length) [nucleotides]", title = "Almost all ORF containing transcripts are longer than the median", subtitle = "Histogram of log transformed transcript length") + geom_vline(aes(xintercept = log(median(total_length))), color = "red") + facet_wrap(~rna_type, scales = "free_y")
```



## Antisense transcripts


These sequences were then aligned back to the whole transcriptome with blastn. The resulting alignments were filtered to be plus/minus alignments and to span longer than 300 bp. The Target transcripts thus found are labelled as antisense transcripts, because they match the protein coding mRNAs at least partially on their antisense strand. Out of the 39,844 unique transcripts with a swissprot annotation, we find only 9930 with at least one antisense transcript. On the other hand, the total number of unique antisense transcripts found this way is 26,953.


The length distribution of the two differ, with the antisense transcripts being closer to the whole transcriptome numbers (i. e. skewing towards smaller). 

```{r summary, message=FALSE, warning=TRUE}
select(ampirr, total_length.qer, total_length.target) %>% transmute(Protein_coding_mRNAs = total_length.qer, Antisense_Transcripts = total_length.target) %>% summary()
```
We can see that through all quartiles, Protein coding mRNAs are longer than antisense Transcripts, with the median protein coding mRNA being 1kb while half of antisense transcripts are shorter than 413 nucleotides (this is after already filtering all non coding RNAs shorter than 200 bp). Compare to the total length distribution of the transcriptome: 75% of all transcripts have a length of less than 687 nucleotides:

```{r summary whole, echo=FALSE, message=FALSE, warning=FALSE}
summary(longitudes)
```



```{r Grafica de longitud, message=FALSE, warning=FALSE}
#pivot to longer dataframe
ampirr %>% transmute(qseqid, sseqid, Protein_coding_mRNAs = total_length.qer, Antisense_Transcripts = total_length.target) %>% 
  pivot_longer(cols = Protein_coding_mRNAs:Antisense_Transcripts, names_to = "mRNA_type", values_to = "mRNA_length") %>% 
  #This operation is to chose the transcript id from the two seqid columns
  transmute(transcript_id = if_else(mRNA_type == "Protein_coding_mRNAs", qseqid, sseqid), mRNA_type, mRNA_length) %>% unique() %>% 
  #now you can plot
  ggplot(aes(x = log(mRNA_length))) + geom_histogram() + scale_x_continuous(n.breaks = 10) + theme_bw() + facet_wrap(~mRNA_type, nrow = 2, scales = "free_y") + labs(x = "log of mRNA length [nucleotides]", title = "Antisense transcripts are generally shorter than their targets, but more numerous", subtitle = "Histogram distribution of transcript length")

```



